src/Razor/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -3152,6 +3152,391 @@ public class MyComponent<TItem> : ComponentBase
             CompileToAssembly(generated);
         }
 
+        [Fact]
```

Comment:
We might want to have a test that uses `splatting` as well as other features like `@key` and so on to validate they are integrated correctly.

It might also be interesting to have a test where we simulate a side effect to make sure the order is preserved.

`<Comp Something1="object.SideEffect()"  Something2="object" />`

Reply:
I'm really skeptical about whether it adds value to snapshot the code generated for an unrelated combination of features, as it's very hard to reason about. We have to be cautious about doing this too much if we want to be able to manage the whole test suite and be able to say with confidence that the baselines mean something and when they should or shouldn't change. `@key` and splatting are really orthogonal to cascaded generic type inference.

Nonetheless I'm willing to add one such test here and have done so.

> It might also be interesting to have a test where we simulate a side effect to make sure the order is preserved.

All of the tests do that, in that they are showing the exact order of evaluation when there are multiple parameters.